packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -36,6 +38,18 @@ export const useSignUpContext = (): SignUpContextType => {
   const { queryParams } = useRouter();
   const options = useOptions();
 
+  const initialValuesFromQueryParams = useMemo(() => {
+    const props: Record<string, string> = {};
+    const searchParams = new URLSearchParams(window.location.search);
```

Comment:
❓ Is it safe to access the global `window` object here?

Reply:
Can't we access search params from the router ?

Reply:
Uhm yeah we can use `router.querystring`. I'll use that.